src/main/java/com/microsoft/jenkins/function/commands/ZipDeployCommand.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.jenkins.function.commands;
+
+import com.microsoft.azure.management.appservice.WebAppBase;
+import com.microsoft.jenkins.azurecommons.command.CommandState;
+import com.microsoft.jenkins.azurecommons.command.IBaseCommandData;
+import com.microsoft.jenkins.azurecommons.command.ICommand;
+import com.microsoft.jenkins.azurecommons.telemetry.AppInsightsUtils;
+import com.microsoft.jenkins.function.AzureFunctionPlugin;
+import com.microsoft.jenkins.function.util.Constants;
+import hudson.FilePath;
+import hudson.Util;
+import hudson.util.DirScanner;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class ZipDeployCommand implements ICommand<ZipDeployCommand.IZipDeployCommandData> {
+    private static final String ZIP_FOLDER_NAME = "fileArchive";
+    private static final String ZIP_NAME = "archive.zip";
+    private static final String LOCAL_SETTINGS_FILE = "local.settings.json";
+
+    @Override
+    public void execute(IZipDeployCommandData context) {
+        FilePath workspace = context.getJobContext().getWorkspace();
+        String filePattern = Util.fixNull(context.getFilePath());
+
+        final FilePath tempDir;
+        try {
+            tempDir = workspace.createTempDir(ZIP_FOLDER_NAME, null);
+            final FilePath zipPath = tempDir.child(ZIP_NAME);
+            final DirScanner.Glob globScanner = new DirScanner.Glob(filePattern, excludedFilesAndZip());
+            workspace.zip(zipPath.write(), globScanner);
+
+            WebAppBase functionApp = context.getWebAppBase();
```

Comment:
null check?

Reply:
If the app is null, will throw an exception when the process starts.